editor/gui/scene_tree_editor.cpp

```diff
@@ -174,15 +174,42 @@ void SceneTreeEditor::_cell_button_pressed(Object *p_item, int p_column, int p_i
 		EditorDockManager::get_singleton()->focus_dock(NodeDock::get_singleton());
 		NodeDock::get_singleton()->show_groups();
 	} else if (p_id == BUTTON_UNIQUE) {
-		undo_redo->create_action(TTR("Disable Scene Unique Name"));
-		undo_redo->add_do_method(n, "set_unique_name_in_owner", false);
-		undo_redo->add_undo_method(n, "set_unique_name_in_owner", true);
-		undo_redo->add_do_method(this, "_update_tree");
-		undo_redo->add_undo_method(this, "_update_tree");
-		undo_redo->commit_action();
+		bool ask_before_revoking_unique_name = bool(EDITOR_GET("docks/scene_tree/ask_before_revoking_unique_name"));
+		if (ask_before_revoking_unique_name) {
+			String msg = vformat(TTR("Revoke Unique Name for node \"%s\"?"), n->get_name());
+			ask_before_revoke_checkbox->set_pressed(false);
+			revoke_dialog_label->set_text(msg);
+
+			if (revoke_dialog->is_connected(SceneStringName(confirmed), callable_mp(this, &SceneTreeEditor::_revoke_unique_name))) {
+				revoke_dialog->disconnect(SceneStringName(confirmed), callable_mp(this, &SceneTreeEditor::_revoke_unique_name));
+			}
+			revoke_dialog->connect(SceneStringName(confirmed), callable_mp(this, &SceneTreeEditor::_revoke_unique_name).bind(n));
+
+			revoke_dialog->reset_size();
+			revoke_dialog->popup_centered();
+		} else {
+			_revoke_unique_name(n);
+		}
+	}
+}
+
+void SceneTreeEditor::_update_ask_before_revoking_unique_name() {
+	if (!ask_before_revoke_checkbox->is_pressed() != bool(EditorSettings::get_singleton()->get_setting("docks/scene_tree/ask_before_revoking_unique_name"))) {
+		EditorSettings::get_singleton()->set("docks/scene_tree/ask_before_revoking_unique_name", !ask_before_revoke_checkbox->is_pressed());
```

Comment:
```suggestion
	if (!ask_before_revoke_checkbox->is_pressed()) {
		EditorSettings::get_singleton()->set("docks/scene_tree/ask_before_revoking_unique_name", false);
		ask_before_revoke_checkbox->set_pressed(false);
```
That's much simpler. The checkbox can't be enabled back, so no reason to check for the other state.